resources/assets/js/annotations/annotatorContainer.vue

```diff
@@ -581,20 +592,20 @@ export default {
     watch: {
         imageId(id) {
             if (id) {
-                this.imageLoadingError = false;
                 this.startLoading();
                 this.crossOriginError = false;
                 Vue.Promise.all(this.getImageAndAnnotationsPromises(id))
                     .then(this.setCurrentImageAndAnnotations)
-                    .then(this.updateUrlSlug)
                     .then(this.maybeUpdateAnnotationMode)
-                    .then(this.emitImageChanged)
                     .then(this.maybeShowTilingInProgressMessage)
-                    // When everything is loaded, pre-fetch the data of the next and
-                    // previous images so they can be switched fast.
-                    .then(this.cachePreviousAndNext)
                     .catch(this.handleLoadingError)
-                    .finally(this.finishLoading);
+                    .finally(() => {
+                        this.updateUrlSlug();
+                        this.emitImageChanged();
+                        // When everything is loaded, pre-fetch the data of the next and
+                        // previous images so they can be switched fast.
+                        this.cachePreviousAndNext();
+                        this.finishLoading()});
```

Comment:
I feel this is getting too convoluted with the additional logic. Maybe it's time to introduce async/await (for the first time) for better readability. How do you like something like this?

```js
async imageId(id) {
    if (!id) {
        return;
    }

    this.startLoading();
    this.crossOriginError = false;

    try {
        let [image, annotations] = await Vue.Promise.all(this.getImageAndAnnotationsPromises(id));
        this.imageLoadingError = false;
        this.image = image;
        this.annotations = annotations;
        this.maybeUpdateAnnotationMode();
        this.maybeShowTilingInProgressMessage();
    } catch (e) {
        if (e instanceof CrossOriginError) {
            this.crossOriginError = true;
        } else {
            this.imageLoadingError = true;
            this.image = null;
            this.annotations = [];
            Messages.danger(e);
        }
    } finally {
        this.updateUrlSlug();
        this.emitImageChanged();
        // When everything is loaded, pre-fetch the data of the next and
        // previous images so they can be switched fast.
        this.cachePreviousAndNext();
        this.finishLoading();
    }
},
```

This makes `setCurrentImageAndAnnotations()` and `handleLoadingError()` unnecessary. Also, I think you can read/understand it better. Or is it just me because I want to use async/await? :smiley: 

This may also have other implications that I will discuss in the other comments.

Reply:
maybeUpdateAnnotationMode() expects a 'data' argument. This was provided by the setCurrentImageAndAnnotations() promise. Now you cannot draw annotations because "[Vue warn]: Error in callback for watcher "isDefaultInteractionMode": "TypeError: this.modifyInteraction is undefined"". 
I don't get why this error occurs because 'data' was undefined even before this change.

Reply:
This is unrelated to the `data` argument. `maybeUpdateAnnotationMode()` can also work without it. I'll respond to the error in my new review.